chore: pypi oidc publish

neo-jesse opened the pull request:
### Migrate PyPI publishing from API tokens to OIDC trusted publishing

Overview:
Replaces long-lived PyPI API tokens with OIDC (OpenID Connect) trusted publishing for enhanced security and tokenless deployments.

Changes:
✅ New OIDC deploy action (.github/actions/python/pypi-deploy-oidc/action.yaml)
✅ Updated workflows with OIDC permissions (id-token: write)
✅ Replaced token secrets with trusted publisher configuration
✅ Added setup documentation (.github/PYPI_OIDC_SETUP.md)

Security Benefits:
🔒 No more stored API tokens - eliminates token leakage risk
🔄 Automatic authentication via GitHub Actions OIDC
📊 Better audit trail for publishing activities
🛡️ Enhanced security with short-lived tokens

Projects Updated:
opentrons (API package)
opentrons-shared-data (shared data package)

Next Steps:
Test with a real release tag
Remove old API token secrets after confirming OIDC works
Breaking Changes: None - maintains same deployment behavior with improved security.

neo-jesse committed:
testing new way of publishing

neo-jesse committed:
Add OIDC PyPI publishing test workflow for pypi-publish branch

neo-jesse committed:
Fix YAML syntax error in OIDC PyPI deploy action

neo-jesse committed:
Add dedicated PyPI upload test workflow

neo-jesse committed:
Clean up test workflows and tags - OIDC setup ready for production

neo-jesse committed:
Merge branch 'edge' into pypi-publish

neo-jesse committed:
Switch to official pypa/gh-action-pypi-publish for OIDC trusted publishing

- Replace custom OIDC action with official pypa/gh-action-pypi-publish@release/v1
- Update API and shared-data workflows to use official action for production PyPI
- Keep token-based publishing for Test PyPI (development/testing)
- Remove custom pypi-deploy-oidc action
- Update documentation to reflect official action usage

neo-jesse committed:
Add test workflow for official PyPI publish action

- Test official pypa/gh-action-pypi-publish action configuration
- Verify workflow setup and OIDC permissions
- Check package building and action availability
- Validate documentation completeness
- Run on pypi-publish branch only

neo-jesse committed:
Focus test on trusted publishing only

- Remove configuration and documentation tests
- Test actual PyPI trusted publishing with OIDC
- Upload to Test PyPI to verify trusted publishing setup
- Provide clear pass/fail results and next steps
- Use continue-on-error to handle expected failures before setup

neo-jesse committed:
Trigger trusted publishing test after PyPI setup

sfoster1 commented on .github/oidc-trust-policy.json:
```diff
@@ -0,0 +1,20 @@
+    {
+      "effect": "Allow",
+      "principal": {
+        "Federated": "arn:aws:iam::ACCOUNT_ID:oidc-provider/token.actions.githubusercontent.com"
```
is this suposed to be the literal string `ACCOUNT_ID`

neo-jesse committed:
Test against Production PyPI where trusted publishing is configured

sfoster1 commented on .github/workflows/test-pypi-official-action.yaml:
```diff
@@ -0,0 +1,85 @@
+          echo "## 🧪 PyPI Trusted Publishing Test Results" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          
+          if [ "${{ job.status }}" == "success" ]; then
```
don't really love this stuff being here. it would be one thing if the multiple outputs were actually based on separate outcome data, like it could tell you if pypi was configured but oidc wasn't, but this is just a lot of text for a binary outcome.

sfoster1 commented on .github/workflows/test-pypi-official-action.yaml:
```diff
@@ -0,0 +1,85 @@
+      
+      - name: 'Build package'
+        run: |
+          echo "🔨 Building API package..."
```
similarly, the action writes the commands its running so the emojis are kind of duplicative

neo-jesse replied:
this was just for the test

sfoster1 commented on .github/PYPI_OIDC_SETUP.md:
```diff
@@ -0,0 +1,99 @@
+# PyPI OIDC Trusted Publishing Setup
```
the stuff in this document IMO is better just in the pull request description - it's mostly a copy of pypi's and the action's official documentation

sfoster1 reviewed (changes requested):
couple little things

neo-jesse committed:
Remove test workflow - trusted publishing is working correctly

- Remove test-pypi-official-action.yaml
- Trusted publishing setup is complete and verified
- Ready for production use

y3rsh committed:
fix(analyses-snapshot): add agent instructions and be less verbose (#19900)

neo-jesse committed:
removed unneeded files

CaseyBatten committed:
feat(api): PAPI capture_image command implementation (#19902)

EXEC-1951
Implements the `capture_image` PAPI command to enable image capture during protocol run.

ryanthecoder committed:
feat(api): add dynamic pipetting to mix (#19883)

# Overview
Add a new version of the mix function that allows the user to do complicated mixes that were previously only possible by looping aspirate and dispense.

mjhuff committed:
feat(robot-server): support downloading all images as zip from /dataFiles/:runId/images/download (#19871)

Closes EXEC-1994

mjhuff committed:
feat(robot-server): Support `DiskMonitor` (#19903)

Works towards EXEC-1963

Adds the DiskMonitor class, which monitors disk space and directory size usage, specifically the images directory now. Threshold values which mark general disk space availability as low or the image directory disk size as high are tunable in settings.py as system_low_space_threshold_mb and images_directory_max_size_mb respectively.

ncdiehl11 committed:
feat(app, protocol-designer, step-generation): wire up formToArgs for tip selection (#19894)

This PR updates args types and wires up tip selection fields in
formToArgs for mix and moveLiquid steps. New fields to be wired up in
formToArgs:
- `tip_tracking`
- `tiprack_selected`
- `tips_selected`

neo-jesse committed:
Merge branch 'edge' into pypi-publish

neo-jesse committed:
formatting

neo-jesse committed:
Merge branch 'edge' into pypi-publish

neo-jesse committed:
don't need that comment

neo-jesse committed:
extra space

neo-jesse committed:
Merge branch 'edge' into pypi-publish

sfoster1 reviewed (approved):
great, nice work!

y3rsh reviewed (approved):
Thank you! Was hoping it could be this easy.
